packages/hash/frontend/src/components/hooks/blockProtocolFunctions/ontology/useBlockProtocolGetPropertyType.ts

```diff
@@ -16,8 +16,14 @@ export const useBlockProtocolGetPropertyType = (): {
     GetPropertyTypeQuery,
     GetPropertyTypeQueryVariables
   >(getPropertyTypeQuery, {
-    // Property types are immutable, any request for an propertyTypeId should always return the same value.
-    fetchPolicy: "cache-first",
+    /**
+     * Property types are immutable, any request for an propertyTypeId should always return the same value.
+     * However, currently requests for non-existent property types currently return an empty subgraph, so
+     * we can't rely on this.
+     *
+     * @todo revert this back to cache-first once that's fixed
+     */
```

Comment:
Requests for a specific entity / type should return an error if it doesn't exist. Right now they just return an empty subgraph. This is a known problem, and this was the temporary patch for getEntityType. Without this change, validation would break when you go to create a property type with the same name as one you just created, as the request returning an empty subgraph for that slug will be cached. 

This temporary fix was agreed with @Alfred-Mountfield and @benwerner01 for getEntityType.

Reply:
I see now, thanks for the explanation!